### ISSUE TITLE
Add support for tags in detectors

### ISSUE THREAD

**Issue by matfra** [splunk-terraform/terraform-provider-signalfx#124]
Tags can be pretty useful to search and organize detectors. In Signalform we support them:
https://github.com/Yelp/terraform-provider-signalform/blob/master/src/terraform-provider-signalform/signalform/detector.go#L89

This is one of the only issues that we've spotted to prevent us to start migrating from signalform to signalfx provider

**Comment by cory-signalfx** [comment]
Hey there @matfra, thanks for the note!

Unfortunately this is problematic for the provider because the SignalFx API for tags is not synchronous with the request. This causes unclean plans when the provider tries to verify the response.

This worked in SignalForm because SignalForm does *not* check the response. Because checking the response is a requirement for being an official Terraform provider it had to be removed.

If we added tags in the current provider it would require adding rather long client side delays and re-requests of the dashboard/detector after the backend has had time to quiesce. This is a long winded way of saying, for now, "as intended" and "wont fix". :(

We've got some tickets on the SignalFx side to get to a fix for this, but until then these are effectively unsafe for Terraform. Sorry for the disappointment. :(
